src/hotspot/os/aix/porting_aix.cpp

```diff
@@ -1049,14 +1050,25 @@ void* Aix_dlopen(const char* filename, int Flags, const char** error_report) {
     return nullptr;
   }
   else {
+    // extract member string if exist and generate hash of it
+    unsigned long hash = 0;
+    const char* substr;
+    if (filename[strlen(filename) - 1] == ')' && (substr = strrchr(filename, '('))) {
+      // Mocklisp hash function.
+      while (*substr) {
+        hash = (hash << 5) - hash + (unsigned long)*substr++;
+      }
+    }
```

Comment:
How likely is a hash collision? My first thought was to store the member string in an additional array. But there is no length limitation. So I cannot just add an char array of e.g. 20 chars, but have to use a pointer to an external dynamically loaded memory holding the string. I came to the conclusion that a hash is much more elegant with a neglectable small risk of a hash collision.

Reply:
I guess extremely unlikely. But if it happens, it will be an extremely annoying problem.
I'd probably use strdup, but let's hear what other reviewers think.